### DISCUSSION TITLE
Grouped fields as field, "component", with automatic scoped relations

### DISCUSSION THREAD

**Discussion by viters** [directus/directus#11601] [Feature Requests]
Strapi does have such feature and it is really useful for managing some reusable content schemas through panel:

https://docs.strapi.io/user-docs/latest/content-types-builder/configuring-fields-content-type.html#components

#### Description

Component is a specific type of collection that is created for each relation. It may be treated as a "field group". Each collection with a field of type "component" can have one, multiple or zero (if not required) instances of it. Moreover, those instances are scoped to one item, so they do not mix with other items' components instances nor other types' items' components instances.

#### Example usage

https://strapi.io/blog/headless-cms-strapi-seo-best-practices#:~:text=Let%27s%20name%20the%20component%20as%20Seo%20which%20belongs%20to%20the%20Shared%20category.%20Since%20the%20Shared%20category%20does%20not%20exist%2C%20typing%20it%20in%20the%20category%20form%20field%20will%20create%20it%20for%20us.
(chrome link to part of article)

I can have `seo` component with fields `title`, `description`, `image`. That component may be added as `seo` field to other collections, like `videos`. The `videos_component_seo` database table is created. Relation can be either `o2o` (video can have only one `seo`) or `o2m` (video can have multiple `seo`, not useful here though).

When I edit video, I can access editing the details of `seo`. Specific `seo` instance (or instances in `m2m`) are not visible to other videos objects.

Adding `seo` to another collection, like `events` would produce different table, e.g. `events_component_seo`.

The actual implementation underneath with "table per relation per component" may not be the best solution - it is just a proposal to present the idea.

#### Current state in Directus

AFAIK, achieving smooth management of such data is problematic in Directus right now. We have to create multiple collections manually and duplicate their schemas or use one collection and m2m/o2m relation - both solutions have the problem of picking proper instance of component or scoping that instance to specific object. Either we can duplicate the fields on target collection (duplicate `seoTitle`, `seoDescription`, `seoImage` for each collection that requires them).

**Comment by jamescammarano** [comment]
Honest question because I'm not sure I follow but would this be solved by/be similar to:
#3414

**Comment by joselcvarela** [comment]
I believe it's not the same thing.
Viters is basically asking for a field that is a group of another fields. For example:
```
address: <-- group field
  line1: <-- regular field inside address
  line2:
  street
  country
  ...
```

Also, Prismic has those Slices which are almost the same.
https://prismic.io/docs/core-concepts/slices

**Comment by rijkvanzanten** [comment]
I might be misreading / misunderstanding this, but isn't this effectively the Many to Any field? 🤔

**Reply by viters** [reply]
In database - yes, it might be. In interface - no.

**Reply by rijkvanzanten** [reply]
Ohh I see what you're saying. Is it basically a preconfigured set of fields you can then quickly insert into other collections for ease of configuration? Kinda like the "templates" mentioned over here https://github.com/directus/directus/discussions/11535

**Comment by pzpsofficial** [comment]
Hi! Any update on this?

**Reply by rijkvanzanten** [reply]
This is an open feature request. Based on community engagement, this may or may not be determined to be implemented as is. 

As of right now, you can already configure a very similar solution by relying on the many to any interface 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
